File: Applied Course/4.Problem Solving/10.Problems on Greedy Algorithms/5.Minimum Number of Arrows to Burst Balloons.py
# https://leetcode.com/problems/minimum-number-of-arrows-to-burst-balloons/
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        if not points:
            return 0
        points.sort()
        arrows=1
        currentEnd=points[0][1]
        for x in points[1:]:
            if x[0]<=currentEnd:
                currentEnd=min(currentEnd,x[1])
            else:
                arrows+=1
                currentEnd=x[1]
        return arrows

File: Applied Course/4.Problem Solving/10.Problems on Greedy Algorithms/5.Minimum Number of Arrows to Burst Balloons.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        if(points.empty())
            return 0;
        sort(points.begin(),points.end());
        int arrows=1;
        int currentEnd=points[0][1];
        for(int i=1;i<points.size();i++)
        {
            if(points[i][0]<=currentEnd)
                currentEnd=min(currentEnd,points[i][1]);
            else
            {
                arrows++;
                currentEnd=points[i][1];
            }
        }
        return arrows;
    }
};
int main()
{
    // https://leetcode.com/problems/minimum-number-of-arrows-to-burst-balloons/
    return 0;
}